crates/rolldown/src/stages/link_stage/mod.rs

```diff
@@ -62,42 +62,47 @@ pub struct LinkStage<'a> {
 
 impl<'a> LinkStage<'a> {
   pub fn new(scan_stage_output: ScanStageOutput, options: &'a SharedOptions) -> Self {
+    let mut errors = scan_stage_output.errors;
+    let metas = scan_stage_output
+      .module_table
+      .modules
+      .iter()
+      .map(|module| LinkingMetadata {
+        dependencies: module
+          .import_records()
+          .iter()
+          .filter_map(|rec| {
+            if options.inline_dynamic_imports || !matches!(rec.kind, ImportKind::DynamicImport) {
+              Some(rec.resolved_module)
+            } else {
+              if matches!(options.format, OutputFormat::Iife) {
+                // Throw an error.
```

Comment:
The rollup check chunks count to give warnings, see https://github.com/rollup/rollup/blob/master/src/Bundle.ts#L68. I think you should check it at code splitting.

Reply:
> Does rollup inlineDynamicImports no matter if inlineDynamicImports enabled while targeting IFFE format?

cc @hyf0 The rollup checked at here.

Reply:
In this scenario, Rollup throws an error (as seen [here](https://github.com/rollup/rollup/blob/master/src/Bundle.ts#L228)).

However, regarding [#1393](https://github.com/rolldown/rolldown/pull/1393#discussion_r1641910564), there's a concern:
> There might be issues with this implementation when working with asset modules and future copy and file loaders (see [#1295](https://github.com/rolldown/rolldown/issues/1295)).

Rollup's approach can be found [here](https://github.com/rollup/rollup/blob/186706b7229943c97e0f2742a2944c6a5d74b0ff/src/Bundle.ts#L228-L244).

I propose implementing this logic at this point and deferring the handling of asset modules to the `file` / `copy` loader implementation. This approach should also apply to the single file checker.